src/validations/rules/sar.sch

```diff
@@ -176,6 +175,140 @@
 
     </sch:pattern>
 
+    <sch:pattern
+        id="results">        
+
+        <sch:rule
+            context="oscal:result">
+            <sch:let
+                name="sap-import-url"
+                value="resolve-uri(/oscal:assessment-results/oscal:import-ap/@href, base-uri())" />
+            <sch:let
+                name="sap-available"
+                value="
+                    if (: this is not a relative reference :) (not(starts-with($sap-import-url, '#')))
+                    then
+                        (: the referenced document must be available :)
+                        doc-available($sap-import-url)
+                    else
+                        true()" />
+            <sch:let
+                name="sap-doc"
+                value="
+                    if ($sap-available)
+                    then
+                        doc($sap-import-url)
+                    else
+                        ()" />
+            <sch:let
+                name="ssp-import-url"
+                value="resolve-uri($sap-doc/oscal:assessment-plan/oscal:import-ssp/@href, base-uri())" />
+            <sch:let
+                name="ssp-available"
+                value="
+                    if (: this is not a relative reference :) (not(starts-with($ssp-import-url, '#')))
```

Comment:
The SSP can be available via an intra-SAP relative reference which requires using the referenced `resource` to obtain (if possible) the SAP via the `resource/rlink`.